src/SmartComponents/BaselinesPage/EditBaseline/helpers.js

```diff
@@ -1,49 +1,155 @@
+import React from 'react';
 import jiff from 'jiff';
+import FactKebab from './FactKebab/FactKebab';
 
-function buildParentFact(rows, parentRowId) {
-    let childRows = [];
-    if (rows.length === 0 || parentRowId === undefined) {
-        return childRows;
+function renderKebab(factName, valueName, fact) {
+    return <td>
+        { <FactKebab
+            factName={ factName }
+            valueName={ valueName }
+            fact={ fact } />
+        }
+    </td>;
+}
+
+function buildNewFactData(isParent, factName, valueName, factData) {
+    let newFactData = {};
+    let subFacts = [];
+
+    if (factData && !Array.isArray(factData)) {
+        factData.values.forEach(function(subFact) {
+            subFacts.push(subFact);
+        });
+        subFacts.push({ name: factName, value: valueName });
+
+        newFactData.name = factData.name;
+        newFactData.values = subFacts;
+
+        return newFactData;
+    }
+
+    newFactData.name = factName;
+    if (isParent) {
+        newFactData.values = [];
     } else {
-        childRows = rows.filter(function(row) {
-            return row.parent === parentRowId;
-        })
-        .map(function(row) {
-            return { name: row.data.modules[0], value: row.data.modules[1] };
+        newFactData.value = valueName;
+    }
+
+    return newFactData;
+}
+
+function buildEditedFactData(isParent, originalFactName, factName, originalValueName, valueName, factData) {
```

Comment:
I've seen this in a few places. What does `valueName` represent?

Reply:
It's the value of the fact. Like `{ name: factName, value: valueName }`

Reply:
If valueName repesents the value of the fact in the key-value sense, I would find it more understandable when parsing the code to have it named something like factValue.